src/Migration.php

```diff
@@ -194,14 +224,16 @@ public function prepareEnvironment()
      */
     public function reset($upVersion = null)
     {
+        $fileInfo = $this->getFileContent($this->getBaseSql());
+
         if ($this->callableProgress) {
-            call_user_func_array($this->callableProgress, ['reset', 0]);
+            call_user_func_array($this->callableProgress, ['reset', 0, $fileInfo]);
         }
         $this->getDbCommand()->dropDatabase();
         $this->getDbCommand()->createDatabase();
         $this->getDbCommand()->createVersion();
 
-        if (file_exists($this->getBaseSql())) {
+        if ($fileInfo["exists"]) {
             $this->getDbCommand()->executeSql(file_get_contents($this->getBaseSql()));
```

Comment:
You don't need to perform this `file_get_contents` anymore, you can replace it with `$fileInfo["content"]`
```suggestion
            $this->getDbCommand()->executeSql($fileInfo["content"]);
```